File: app/src/main/java/com/github/weatherapp/data/OpenWeatherApi.java
package com.github.weatherapp.data;

import com.github.weatherapp.core.CurrentForecast;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface OpenWeatherApi {

    @GET("weather")
    Call<CurrentForecast> getCurrentWeatherForecast(@Query("appid") String apiKey, @Query("q") String city, @Query("units") String unit);
}

File: app-kotlin/src/main/kotlin/com/github/weatherapp/data/OpenWeatherApi.kt
package com.github.weatherapp.data

import com.github.weatherapp.core.CurrentForecast

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface OpenWeatherApi {

    @GET("weather")
    fun getCurrentWeatherForecast(@Query("appid") apiKey: String, @Query("q") city: String, @Query("units") unit: String): Call<CurrentForecast>
}
